packages/openapi3/test/test-openapi-output.ts

```diff
@@ -378,6 +378,276 @@ describe("openapi3: definitions", () => {
+      type: "object",
+      properties: {},
+      discriminator: { propertyName: "kind" },
+      oneOf: [{ $ref: "#/components/schemas/Cat" }, { $ref: "#/components/schemas/Dog" }],
+    });
+  });
+
+  it("defines discriminated unions with non-empty base type", async () => {
+    const openApi = await openApiFor(`                                                             
+      @discriminator("kind")                                                                       
+      model Pet {
+        name: string;
+        weight?: float32;
+      }                                                                                
+      model Cat extends Pet {                                                                      
+        kind: string;                                                                              
+        meow: int32;                                                                               
+      }                                                                                            
+      model Dog extends Pet {                                                                      
+        kind: string;                                                                              
+        bark: string;                                                                              
+      }                                                                                            
+                                                                                                    
+      @route("/")                                                                                  
+      namespace root {                                                                             
+        op read(): { @body body: Pet };                                                            
+      }                                                                                            
+      `);
+    ok(openApi.components.schemas.Cat, "expected definition named Cat");
+    ok(openApi.components.schemas.Dog, "expected definition named Dog");
+    ok(openApi.components.schemas.Pet, "expected definition named Pet");
+    deepStrictEqual(openApi.paths["/"].get.responses["200"].content["application/json"].schema, {
+      $ref: "#/components/schemas/Pet",
+    });
+    deepStrictEqual(openApi.components.schemas.Pet, {
+      type: "object",
+      properties: { name: { type: "string" }, weight: { type: "number", format: "float" } },
+      required: ["name"],
+      discriminator: { propertyName: "kind" },
+      oneOf: [{ $ref: "#/components/schemas/Cat" }, { $ref: "#/components/schemas/Dog" }],
+    });
+  });
+
+  it("defines discriminated unions with discriminator property in base type", async () => {
+    const openApi = await openApiFor(`                                                             
+      @discriminator("kind")                                                                       
```

Comment:
don't feel like we want this - for a usable api, we have to be able to count on the discriminator values being known and unique